pkg/forwarder/forwarder_health_test.go

```diff
@@ -29,28 +29,35 @@ func TestHasValidAPIKey(t *testing.T) {
 		ts2.URL: {"key3"},
 	}
 
-	fh := forwarderHealth{keysPerDomains: keysPerDomains}
-	fh.init()
+	fh := newForwarderHealth(keysPerDomains)
 	assert.True(t, fh.hasValidAPIKey())
 
 	assert.Equal(t, &apiKeyValid, apiKeyStatus.Get("API key ending with _key1"))
 	assert.Equal(t, &apiKeyValid, apiKeyStatus.Get("API key ending with _key2"))
 	assert.Equal(t, &apiKeyValid, apiKeyStatus.Get("API key ending with key3"))
 }
-
-func TestComputeDomainsURL(t *testing.T) {
+func TestComputeAPIDomains(t *testing.T) {
 	keysPerDomains := map[string][]string{
-		"https://app.datadoghq.com": {"api_key1"},
+		"https://app.datadoghq.com":    {"api_key_1"},
+		"https://app.datadoghq.eu":     {"api_key_2"},
+		"https://app.datad0g.com":      {"api_key_3"},
+		"https://app.datad0g.eu":       {"api_key_4"},
```

Comment:
some of these cases can be moved to `TestComputeAPIDomain` in `config_test`, and it'd be interesting to check what happens with a URL that has more than 1 api key